### PR TITLE
gitlab modules: minor refactor

### PR DESCRIPTION
**Pull request by russoz**
##### SUMMARY

Refactor function `vars_to_variables` used in both modules `gitlab_group_variable` and `gitlab_project_variable` to the `gitlab` module utils.

##### ISSUE TYPE

- Refactoring Pull Request

##### COMPONENT NAME
plugins/module_utils/gitlab.py
plugins/modules/gitlab_group_variable.py
plugins/modules/gitlab_project_variable.py

### CODE REVIEW

File: plugins/module_utils/gitlab.py [hunk 1]
```diff
@@ -10,6 +10,8 @@
 
 from ansible.module_utils.basic import missing_required_lib
 from ansible.module_utils.common.text.converters import to_native
+from ansible.module_utils.six import string_types
+from ansible.module_utils.six import integer_types
```
Comment (felixfontein) [inline plugins/module_utils/gitlab.py:14]:
```suggestion
from ansible.module_utils.six import integer_types, string_types
```

File: plugins/module_utils/gitlab.py [hunk 2]
```diff
@@ -121,3 +123,39 @@ def filter_returned_variables(gitlab_variables):
...
                 item.pop(key)
     return existing_variables
+
+
+def vars_to_variables(vars, module):
+    # transform old vars to new variables structure
+    variables = list()
+    for item, value in vars.items():
+        if (isinstance(value, string_types) or
+           isinstance(value, (integer_types, float))):
```
Comment (felixfontein) [inline plugins/module_utils/gitlab.py:133]: Why not just
```suggestion
        if (isinstance(value, (string_types, integer_types, float)):
```
?

File: changelogs/fragments/6384-gitlab-refactor.yml [hunk 3]
```diff
@@ -0,0 +1,3 @@
+minor_changes:
+  - gitlab module utils - refactor function in from modules (https://github.com/ansible-collections/community.general/pull/6384).
```
Comment (felixfontein) [inline changelogs/fragments/6384-gitlab-refactor.yml:2]: I would probably leave this one away, and just mention changes for the modules.

Comment (russoz) [reply changelogs/fragments/6384-gitlab-refactor.yml]: done!

**Comment by felixfontein** [comment]
Did you try running this code to see whether it still works as expected?

**Comment by russoz** [comment]
No I have not. Feel free to sit this one out of the 6.6.0 release to get more review time.

**Comment by felixfontein** [comment]
@markuman is your thumb up for the PR some kind of review? :)

**Review by markuman — APPROVED** [review]

**Comment by markuman** [comment]
I can checkout the PR tomorrow and run it in the wild. But it looks good to me.

**Comment by felixfontein** [comment]
@markuman would be great if you could do that!

**Comment by markuman** [comment]
I made some short tests. Works as expected.

**Comment by felixfontein** [comment]
@russoz thanks for the refactoring!
@markuman thanks for reviewing and testing!

**Pull request merged**